test/manifold_test.cpp

```diff
@@ -586,6 +586,1038 @@ TEST(Manifold, PinchedVert) {
+               0.0,
+               0.0,
+               1.0,
+               0.16666667163372,
+               0.0,
+               1.0,
+               0.0,
+               0.0,
+               0.0,
+               1.0,
+               0.0,
+               1.01398992538452,
+               1.01145005226135,
+               0.0,
+               0.0,
+               0.0,
+               1.0,
+               0.16666667163372,
+               0.0,
+               1.0,
+               0.0,
+               0.0,
+               0.0,
+               0.875,
+               0.0,
+               0.716999173164368,
+               1.01145005226135,
+               0.716999173164368,
+               0.0,
+               0.0,
+               0.875,
+               0.16666667163372,
+               0.716999173164368,
+               1.01145005226135,
+               0.716999173164368,
+               0.0,
+               0.0,
+               0.875,
+               0.16666667163372,
+               0.0,
+               0.0,
+               0.0,
+               0.0,
+               0.0,
+               1.0,
+               0.333333343267441,
+               1.01398992538452,
+               1.01145005226135,
+               0.0,
+               0.0,
+               0.0,
+               1.0,
+               0.16666667163372,
+               0.0,
+               0.0,
+               0.0,
+               0.0,
+               0.0,
+               1.0,
+               0.333333343267441,
+               0.716999173164368,
+               1.01145005226135,
+               0.716999173164368,
+               0.0,
+               0.0,
+               0.875,
+               0.16666667163372,
+               0.0,
+               0.0,
+               0.0,
+               0.0,
+               0.0,
+               0.875,
+               0.333333343267441,
+               0.0,
+               0.0,
+               0.0,
+               0.0,
+               0.0,
+               0.875,
+               0.333333343267441,
+               0.0,
+               1.0,
+               0.0,
+               0.0,
+               0.0,
+               1.0,
+               0.5,
+               0.0,
+               0.0,
+               0.0,
+               0.0,
+               0.0,
+               1.0,
+               0.333333343267441,
+               0.0,
+               1.0,
+               0.0,
+               0.0,
+               0.0,
+               1.0,
+               0.5,
+               0.0,
+               0.0,
+               0.0,
+               0.0,
+               0.0,
+               0.875,
+               0.333333343267441,
+               0.0,
+               1.0,
+               0.0,
+               0.0,
+               0.0,
+               0.875,
+               0.5};
+  complex.vertProperties = vertProps;
+
+  std::vector<uint32_t> triVerts;
+  for (int i = 0; i < 144; i++) {
+    triVerts.push_back(i);
+  }
+  complex.triVerts = triVerts;
+
+  Manifold manifold(complex);
```

Comment:
It looks like you forgot `complex.Merge()` which will be necessary for an STL-style mesh that has no shared verts (as non-manifold as it comes). Did this garbage mesh come out of your revolve function? You're writing each vertex about six times.

Reply:
Yeah! I didn't want to dig too deep into that function to hasten the work.

Reply:
MeshGL64 fails even with `complex.Merge();`